tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -0,0 +1,287 @@
+location of a single RNTuple. The RNTuples are subsequently processed in the order in which they were provided.
+
+The default model of the RNTuple that is provided first will be used to construct the fields in the entry that will be
+filled in each iteration. This entry is owned by the RNTupleProcessor.
+If subsequently processed RNTuples contain additional fields, they will be ignored.
+Conversely, if a field that was present in the first RNTuple is not found in a subsequent one, an error will be thrown.
+
+The object returned by the RNTupleProcessor iterator is a view on the current state of the processor, and provides
+access to the global entry index (i.e., the entry index taking into account all processed ntuples), local entry index
+(i.e. the entry index for only the currently processed ntuple), the index of the ntuple currently being processed (with
+respect to the order of provided RNTupleSpecs) and the actual REntry containing the values for the current entry.
+*/
+// clang-format on
+class RNTupleProcessor {
+private:
+   // clang-format off
+   /**
+   \class ROOT::Experimental::RNTupleProcessor::RFieldContext
+   \ingroup NTuple
+   \brief Manager for a field as part of the RNTupleProcessor.
+
+   An RFieldContext contains two fields: a proto-field which is not connected to any page source but serves as the
+   blueprint for this particular field, and a concrete field that is connected to the page source currently connected
+   to the RNTupleProcessor object for reading. When a new page source is connected, the concrete field gets reset by
+   cloning the proto-field and connecting it to the new page source.
+
+   Apart from the fields themselves, the RFieldContext object also manages the pointer to the value for this particular
+   field that's read by the processor.
+   */
+   // clang-format on
+   class RFieldContext {
+   private:
+      std::unique_ptr<RFieldBase> fProtoField;
+      std::unique_ptr<RFieldBase> fConcreteField;
+      REntry::RFieldToken fToken;
+      std::shared_ptr<void> fValuePtr;
+
+   public:
+      RFieldContext(std::unique_ptr<RFieldBase> protoField, REntry::RFieldToken token)
+         : fProtoField(std::move(protoField)), fToken(token)
+      {
+      }
+
+      RFieldBase &GetProtoField() const { return *fProtoField; }
+      void ResetConcreteField() { fConcreteField = nullptr; }
+      void SetConcreteField() { fConcreteField = fProtoField->Clone(fProtoField->GetFieldName()); }
+      RFieldBase &GetConcreteField() const { return *fConcreteField; }
+      const REntry::RFieldToken &GetToken() const { return fToken; }
+      std::shared_ptr<void> GetValuePtr() const { return fValuePtr; }
+      void SetValuePtr(std::shared_ptr<void> ptr) { fValuePtr = ptr; }
+   };
+
+   // clang-format off
+   /**
+   \class ROOT::Experimental::RNTupleProcessor::RProcessorView
+   \ingroup NTuple
+   \brief View on the RNTupleProcessor iterator state.
+   */
+   // clang-format on
+   class RProcessorView {
```

Comment:
This may be confused with the RNTuple views. Since the view is strictly used in the context of the iterator, how about making it an `RValue` sub class of the `RIterator`?